### PR TITLE
Don't access CookieContainer unless needed

### PR DESCRIPTION
**Pull request by BrennanConroy**
While experimenting with blazor-webassembly and the SignalR C# Client I found that this is the only blocking (not-user fixable) issue for consuming the client with the LongPolling transport.

We set the `_httpConnectionOptions.Cookies` property to a new object in the `HttpConnectionOptions` constructor, so by default it isn't null and will access the `httpClientHandler.CookieContainer` property which will throw in mono wasm.

We want this for 3.1 because blazor-wasm is scheduled for May 2020 and we would like the SignalR Client to work at that time.

No tests because this isn't unit-testable without making a full end-to-end blazor-wasm + signalr test suite.

@Pilchie for 3.1 approval

### CODE REVIEW

File: src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs
Comment (anurse) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]:
```suggestion
                // if the user has configured those options
```

Since it's not just certs now ;)

File: src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs
Comment (anurse) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]:
```suggestion
                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException
```

a) it's "SKUs" not "skews" but..
b) "SKUs" is a wonky internal term, might as well just say variants :)

**Comment by Pilchie** [comment]
Approved for 3.1.0-preview3, pending CI build and code review approvals

**Comment by anurse** [comment]
I'm glad it was a pretty straightforward set of changes needed!

@BrennanConroy can you add a Blazor WASM sample to the repo?

**Comment by BrennanConroy** [comment]
> @BrennanConroy can you add a Blazor WASM sample to the repo?

I'm going to vote for no right now. There is a lot of churn with the HttpClientHandler right now see: https://github.com/aspnet/AspNetCore/pull/16631 so this would break as soon as I made it

**Comment by halter73** [comment]
How afraid are we of temporarily broken samples are we really? If you already have a working sample, I say check it in. Then when the API stabilizes, we can fix the sample instead of creating a new one from scratch.

**Comment by BrennanConroy** [comment]
I tried to add the sample, but it appears to be non-trivial and I wont have any time to try until after preview3.

Please approve (assuming the change is good :D)

**Review by halter73 — APPROVED** [review]

File: src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs
Comment (halter73) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]: Based on the old code it looks like `_httpConnectionOptions` is never null. Was this meant to be the following?

```suggestion
                if (_httpConnectionOptions.Cookies?.Count > 0)
```

**Comment by BrennanConroy** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=410635&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/3261
https://github.com/aspnet/AspNetCore-Internal/issues/3241

**Pull request merged**
